lib/IDE/CompletionLookup.cpp

```diff
@@ -2166,6 +2167,13 @@ void CompletionLookup::foundDecl(ValueDecl *D, DeclVisibilityKind Reason,
     }
 
     return;
+  case LookupKind::StoredProperty:
+    if (auto *VD = dyn_cast<VarDecl>(D)) {
+      if (VD->getImplInfo().hasStorage()) {
+        addVarDeclRef(VD, Reason, dynamicLookupInfo);
+      }
```

Comment:
@xedin I copied this check from https://github.com/apple/swift/blob/cae97a8426910aad471d443c6be23aa88e576dc5/lib/Sema/TypeCheckAttr.cpp#L7323. Do you know/remember why that’s using `VD->getImplInfo().hasStorage()` and not `VD->hasStorage()`?

Reply:
`InitAccessorReferencedVariablesRequest` has to be very careful about the requests it triggers to avoid circularity, in this case I wouldn't think that matters match so you probably fine using `VD->hasStorage()` as @hamishknight suggested.

Reply:
Switched it and uncovered one discrepancy between the two while doing so.

https://github.com/apple/swift/pull/68603/commits/8550799eba4263201c618a2d63b9c4945ce65f93

Reply:
It’s always better to use top-level API when possible to avoid surprises :)